### ISSUE TITLE
 Please explain why remove the mode parameter from RSA operation functions?

### ISSUE THREAD

**Issue by realbrotha** [ARMmbed/mbedtls#5361]
i'm Sorry for choosing the wrong bulletin board.

I wanted to post a question, but I don't have that option, so I'm posting it here.

The rsa_encryption parameter "int mode" disappeared in Mbedtls 3.0 and later, I wonder why.

Are there any vulnerabilities?

**Comment by minosgalanakis** [comment]
Hi,

Would it be possible to elaborate on why there is no option to ask the question on the mailing list? Have you tried subscibing and it did not work?

**Comment by realbrotha** [comment]
I can choose three options. "Bug report", "Enhancement request", "Report a security vulnerability"
These three are not categories of questions.
I wonder why the int mode parameter was deleted from the "mbedtls_rsa_rsaes_pkcs1_v15_encrypt"(rsa.c/.h) method.
The reason for deletion is not explained in the release note. Please explain the reason.

**Comment by mpg** [comment]
> I can choose three options. "Bug report", "Enhancement request", "Report a security vulnerability"

I can see a 4th line with " Mbed TLS mailing list - Mbed TLS community support and general discussion." If you click the "Open" button on that line you should get to the page with information about the mailing list. Alternatively, here's a [direct link](https://lists.trustedfirmware.org/mailman/listinfo/mbed-tls).

I'm closing this issue; please feel free to ask your question on the list and thanks for your interest in Mbed TLS.

**Comment by gilles-peskine-arm** [comment]
In Mbed TLS 2.x, it was deprecated to pass anything other than the “normal” value for `mode`. In Mbed TLS 3.0, we removed deprecated features, including the possibility to pass the wrong value in `mode`. Since only one value of `mode` was now accepted, we removed the `mode` parameter.

Passing the wrong `mode`, i.e. using a public key for a decryption/signature calculation or a private key for an encryption/verification calculation, doesn't make sense from a cryptographic point of view. That's why it was deprecated.

(It's actually possible to use RSA with a pair of private keys which can be used in either direction, i.e. there are two large private exponents rather than a small public one and a large private one, but Mbed TLS has never supported that: it considers `E` to be non-sensitive and so doesn't protect it from side channel attacks.)
